### ISSUE TITLE
`cvCovEst` is very slow for large datasets

### ISSUE THREAD

**Issue by PhilBoileau** [PhilBoileau/cvCovEst#28]
`cvCovEst` is quite slow for relatively large datasets. This is likely due to the observations-based Frobenius loss function that is currently employed. It requires that, for each fold, the cross product of every element in the validation set be computed and summed. This is crippling for large `n` and `p`. Based on Proposition 1 of the manuscript, however, we know that the estimator selection procedure based on this loss is equivalent to that of the matrix-based Frobenius loss. The matrix-based Frobenius loss does not require that the cross product of every element in the validation set be computed, only the validation set's sample covariance matrix.

We should implement this loss as a fold function to pass to `origami::cross_validate`. We might also consider setting it as the default loss function, recommending in the manuscript that it be used in practice for computational reasons.
